### PR TITLE
URL Rewriting topic update

### LINKED ISSUE #7260: Method-based URL rewriting
Any examples of URL rewriting using the method-based approach? I'm interested on how it should be properly done.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 7f5f7c61-a9fa-bd34-001d-217477874bb6
* Version Independent ID: 9c48da92-af90-3fc1-fde0-c3f74fa88510
* Content: [URL Rewriting Middleware in ASP.NET Core](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/url-rewriting?view=aspnetcore-2.1&tabs=aspnetcore2x#additional-resources)
* Content Source: [aspnetcore/fundamentals/url-rewriting.md](https://github.com/aspnet/Docs/blob/master/aspnetcore/fundamentals/url-rewriting.md)
* Product: **aspnet-core**
* GitHub Login: @guardrex
* Microsoft Alias: **riande**

### PR DESCRIPTION
**Pull request by guardrex**
Fixes #7260 

[Internal Review Topic](https://review.docs.microsoft.com/en-us/aspnet/core/fundamentals/url-rewriting?view=aspnetcore-1.0&branch=pr-en-us-9575)

* Big UE pass ... this one has only been lightly updated since it was written. API links ... latest styles applied ... 🌷 table beautification 🌹 ... *the works!*
* Enhancement via feedback from @raberana ... and we did it with nine years to spare! :tada: :smile:
* Takes the samp to 2.2. I'll note it on the sample update tracking issue.
* Linking to 1.1 PDF. Note that without the metadata moniker range, there's no indication that the topic's content is 2.1+ on a <2.1 version load.

### CODE REVIEW

File: aspnetcore/fundamentals/url-rewriting/samples/2.x/SampleApp/README.md [hunk 1]
```diff
@@ -1,8 +1,8 @@
 # ASP.NET Core URL Rewriting Sample (ASP.NET Core 2.x)
 
-This sample illustrates usage of ASP.NET Core 2.x URL Rewriting Middleware. The application demonstrates URL redirect and URL rewriting options. For the ASP.NET Core 1.x sample, see [ASP.NET Core URL Rewriting Sample (ASP.NET Core 1.x)](https://github.com/aspnet/Docs/tree/master/aspnetcore/fundamentals/url-rewriting/samples/1.x).
+This sample illustrates usage of ASP.NET Core 2.x URL Rewriting Middleware. The app demonstrates URL redirect and URL rewriting options.
 
-When running the sample, a response will be served that shows the rewritten or redirected URL when one of the rules is applied to a request URL.
+When running the sample, a response is served that shows the rewritten or redirected URL when one of the rules is applied to a request URL, except with the redirecting an XML file request and rewriting a text file request examples. For the file examples, Static File Middleware serves the file after the request URL is rewritten by the middleware.
```
Comment (mikaelm12) [inline aspnetcore/fundamentals/url-rewriting/samples/2.x/SampleApp/README.md]: >"is applied to a request URL, except with the redirecting an XML ..."

nit: This sentence feels very long. I'd break this up into something like
is applied to a request URL. This though doesn't apply to examples redirecting an XML ..." or something like that.

File: aspnetcore/fundamentals/url-rewriting/samples/2.x/SampleApp/README.md [hunk 2]
```diff
@@ -21,25 +21,30 @@ When running the sample, a response will be served that shows the rewritten or r
 * `Add(RedirectXMLRequests)`
   - Success status code: 301 (Moved Permanently)
   - Example (redirect): **/file.xml** to **/xmlfiles/file.xml**
+* `Add(RewriteTXTRequests)`
```
Comment (mikaelm12) [inline aspnetcore/fundamentals/url-rewriting/samples/2.x/SampleApp/README.md]: nit: I'd consider changing "TXT" to something clearer. I wasn't sure at first what you meant by it when I read it as part RewriteTXTRequests. Just looks a little odd to me, but maybe clear to others

File: aspnetcore/fundamentals/url-rewriting/samples/2.x/SampleApp/Program.cs [hunk 3]
```diff
@@ -16,6 +16,10 @@ public static void Main(string[] args)
             WebHost.CreateDefaultBuilder(args)
                 .UseKestrel(options =>
                 {
+                    // The following option to use port 443 only works if port 
+                    // 443 hasn't already been assigned by the server or otherwise
```
Comment (mikaelm12) [inline aspnetcore/fundamentals/url-rewriting/samples/2.x/SampleApp/Program.cs]: nit: I think you're missing an "is" after "the server or"

File: aspnetcore/fundamentals/url-rewriting.md [hunk 4]
```diff
@@ -3,129 +3,121 @@ title: URL Rewriting Middleware in ASP.NET Core
...
 
-When redirecting requests to a different URL, you indicate whether the redirect is permanent or temporary. The 301 (Moved Permanently) status code is used where the resource has a new, permanent URL and you wish to instruct the client that all future requests for the resource should use the new URL. *The client may cache the response when a 301 status code is received.* The 302 (Found) status code is used where the redirection is temporary or generally subject to change, such that the client shouldn't store and reuse the redirect URL in the future. For more information, see [RFC 2616: Status Code Definitions](https://www.w3.org/Protocols/rfc2616/rfc2616-sec10.html).
+When redirecting requests to a different URL, you indicate whether the redirect is permanent or temporary:
```
Comment (mikaelm12) [inline aspnetcore/fundamentals/url-rewriting.md]: nit: "redirect is permanent or temporary" + "via the status code"

**Comment by guardrex** [comment]
Thanks @mikaelm12 ... see if the last commit fixes things up.

**Comment by mikaelm12** [comment]
Just did another quick scan. Looks good 👍

**Pull request merged**